File: SWEA/p1206.java
import java.util.Scanner;

class Solution
{
	public static void main(String[] args) throws Exception
	{
		Scanner sc = new Scanner(System.in);
		int T;
		T=sc.nextInt();
		for(int test_case = 1; test_case <= T; test_case++)
		{
			int n = sc.nextInt();
			int answer = 0;
			int[] arr = new int[n];
			for(int i=0; i<n; i++) {
				arr[i] = sc.nextInt();
			}
			for(int i=2; i<n-3; i++) {
				int cur = arr[i];
				int leftMax = Math.max(arr[i-2], arr[i-1]);
				int rightMax = Math.max(arr[i+1], arr[i+2]);
				if(cur > leftMax && cur > rightMax) {
					answer += (cur - Math.max(leftMax, rightMax));
				}
			}
			System.out.println("#"+test_case+" "+answer);
		}
	}
}

File: SWEA/p1206.py
T = 10
for test_case in range(1, T + 1):
    n = int(input())
    m = list(map(int, input().split()))
    answer = 0
    for i in range(2, len(m) - 2):
        maxH = max(m[i - 2], m[i - 1], m[i + 1], m[i + 2])
        if m[i] > maxH:
            answer += (m[i] - maxH)
    print(f"#{test_case} {answer}")